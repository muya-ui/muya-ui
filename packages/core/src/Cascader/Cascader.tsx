import React, { Ref, useMemo, useRef } from 'react';

import { useForkRef } from '@muya-ui/utils';

import Trigger from '../Trigger';
import memoForwardRef from '../utils/memoForwardRef';
import useStyles from '../utils/useStyles';
import CascaderInput from './CascaderInput';
import CascaderPanel from './CascaderPanel';
import { ICascaderProps } from './types';
import useCascader from './useCascader';

const defaultStyles = {
  inputWrapper: '',
};

const Cascader = memoForwardRef((props: ICascaderProps, ref: Ref<HTMLDivElement>) => {
  const { styles, inputRef, className, style, children } = props;
  const pickStyles = useMemo(() => {
    if (styles) {
      const { inputWrapper } = styles;
      return { inputWrapper };
    }
  }, [styles]);
  const innerStyles = useStyles('cascader', defaultStyles, pickStyles);
  const innerInputRef = useRef<HTMLInputElement>(null);
  const handleInputRef = useForkRef(inputRef, innerInputRef);
  const { inputProps, panelProps, triggerProps } = useCascader(props, innerInputRef);
  return (
    <Trigger {...triggerProps} popup={<CascaderPanel styles={styles} {...panelProps} />}>
      {React.isValidElement(children) ? (
        children
      ) : (
        <CascaderInput
          ref={ref}
          inputRef={handleInputRef}
          className={[innerStyles.inputWrapper.className, className].join(' ').trim()}
          style={{ ...innerStyles.inputWrapper.style, ...style }}
          {...inputProps}
        />
      )}
    </Trigger>
  );
});

(Cascader as any).__MUYA_CASCADER = true;

export default Cascader;
