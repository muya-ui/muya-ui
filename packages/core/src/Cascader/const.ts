import { ICascaderInputBaseProps } from './types';

export const allCascaderInputProps: Array<keyof ICascaderInputBaseProps> = [
  'maxTagTextLength',
  'maxVerticalTagCount',
  'maxTagWidth',
  'expandIcon',
  'inputValue',
  'popupVisible',
  'multiple',
  'fieldNames',
  'onClear',
  'onRemoveTag',
  'width',
];
