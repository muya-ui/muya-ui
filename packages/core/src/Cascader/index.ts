export { default } from './Cascader';

export { default as Cascader } from './Cascader';
export * from './Cascader';
export * from './types';
