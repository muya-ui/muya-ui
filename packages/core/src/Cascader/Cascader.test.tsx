import React from 'react';
import renderer from 'react-test-renderer';
import mockConsole from 'test/utils/mockConsole';

import { cleanup } from '@testing-library/react';

import Cascader from './Cascader';

const options = [
  {
    label: '福建',
    value: 'fj',
    children: [
      {
        label: '福州',
        value: 'fuzhou',
        children: [
          {
            label: '马尾',
            value: 'mawei',
          },
        ],
      },
      {
        label: '泉州',
        value: 'quanzhou',
      },
    ],
  },
  {
    label: '浙江',
    value: 'zj',
    children: [
      {
        label: '杭州',
        value: 'hangzhou',
        children: [
          {
            label: '余杭',
            value: 'yuhang',
          },
          {
            label: '江干',
            value: 'jianggan',
          },
        ],
      },
    ],
  },
  {
    label: '北京',
    value: 'bj',
    children: [
      {
        label: '朝阳区',
        value: 'chaoyang',
      },
      {
        label: '海淀区',
        value: 'haidian',
        disabled: true,
      },
    ],
  },
];

describe('Cascader', () => {
  beforeAll(() => {
    mockConsole.restoreError();
    mockConsole.mockError();
  });

  afterAll(() => {
    mockConsole.restoreError();
  });

  afterEach(() => {
    cleanup();
  });

  test('测试 Cascader 正常 render', () => {
    const wrapper = renderer
      .create(
        <>
          <Cascader id="test" triggerId="trigger1" inputWidth="80%" options={options} />
          <Cascader triggerId="trigger2" options={options} disabled />
          <Cascader triggerId="trigger3" options={options} disabled>
            <span>[切换城市]</span>
          </Cascader>
          <Cascader
            id="test"
            triggerId="trigger4"
            defaultValue={[['fj', 'quanzhou']]}
            options={options}
            multiple
          />
          <Cascader
            id="test"
            triggerId="trigger5"
            styles={{ inputWrapper: 'custom-input-wrapper' }}
            defaultValue={[['fj', 'quanzhou'], ['bj', 'chaoyang']]}
            options={options}
            multiple
            collapseTags
          />
        </>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
