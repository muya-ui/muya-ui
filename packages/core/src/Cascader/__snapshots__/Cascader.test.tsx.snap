// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cascader 测试 Cascader 正常 render 1`] = `
Array [
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c2 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  width: 100%;
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c4 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(51,51,51,1);
}

.c4::selection {
  background: rgba(26,122,248,0.16);
}

.c4::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c4:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::placeholder {
  color: rgba(204,204,204,1);
}

.c5 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(204,204,204,1);
  cursor: pointer;
}

.c5:hover {
  color: rgba(187,187,187,1);
}

.c5:active {
  color: rgba(153,153,153,1);
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
  width: 80%;
}

.c0 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0 .c1,
.c0 .StyledTagsInputWrapper-kkbZlb {
  cursor: pointer;
}

.c0 .c1 input,
.c0 .StyledTagsInputWrapper-kkbZlb input,
.c0 .c1 svg,
.c0 .StyledTagsInputWrapper-kkbZlb svg {
  cursor: pointer;
}

.c8 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 .c1,
.c9 .StyledTagsInputWrapper-kkbZlb {
  cursor: pointer;
}

.c9 .c1 input,
.c9 .StyledTagsInputWrapper-kkbZlb input,
.c9 .c1 svg,
.c9 .StyledTagsInputWrapper-kkbZlb svg {
  cursor: pointer;
}

.c6 {
  line-height: 0;
  -webkit-animation: bFXLB 150ms linear;
  animation: bFXLB 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

<div
    className="c0 muya-cascader-input-wrapper"
    id="test"
    role="combobox"
    style={Object {}}
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-cascader-input-wrapper muya-input-input-wrapper"
      disabled={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      size="m"
      style={Object {}}
    >
      <input
        aria-describedby={null}
        className="c3 c4 muya-input-input"
        disabled={false}
        id="test"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="请选择"
        readOnly={true}
        size="m"
        type="text"
        value=""
      />
      <div
        className="c5 muya-input-node-wrapper"
        size="m"
      >
        <div
          className="c6"
        >
          <svg
            aria-hidden="false"
            className="c7"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c2 {
  width: 240px;
  height: 32px;
  padding: 0 12px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  line-height: 16px;
  border-radius: 2px;
  border: 1px solid rgba(221,221,221,1);
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  vertical-align: middle;
  background-color: rgba(255,255,255,1);
  background: rgba(250,250,250,1);
  cursor: not-allowed;
}

.c2 input,
.c2 svg {
  cursor: not-allowed;
}

.c4 {
  display: inline-block;
  outline: none;
  border: 0;
  padding: 0;
  margin: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 0;
  background: none;
  font-size: 12px;
  line-height: 16px;
  caret-color: rgba(26,122,248,1);
  color: rgba(221,221,221,1);
  -webkit-text-fill-color: currentColor;
  opacity: 1;
}

.c4::selection {
  background: rgba(26,122,248,0.16);
}

.c4::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c4:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c4::placeholder {
  color: rgba(204,204,204,1);
}

.c5 {
  margin-left: 12px;
  line-height: 0;
  color: rgba(51,51,51,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  outline: none;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(204,204,204,1);
  cursor: pointer;
}

.c5:hover {
  color: rgba(187,187,187,1);
}

.c5:active {
  color: rgba(153,153,153,1);
}

.c8 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c8 .c1,
.c8 .StyledTagsInputWrapper-kkbZlb {
  cursor: pointer;
}

.c8 .c1 input,
.c8 .StyledTagsInputWrapper-kkbZlb input,
.c8 .c1 svg,
.c8 .StyledTagsInputWrapper-kkbZlb svg {
  cursor: pointer;
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 .c3 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c9 .c1,
.c9 .StyledTagsInputWrapper-kkbZlb {
  cursor: pointer;
}

.c9 .c1 input,
.c9 .StyledTagsInputWrapper-kkbZlb input,
.c9 .c1 svg,
.c9 .StyledTagsInputWrapper-kkbZlb svg {
  cursor: pointer;
}

.c6 {
  line-height: 0;
  -webkit-animation: bFXLB 150ms linear;
  animation: bFXLB 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

<div
    className="c0 muya-cascader-input-wrapper"
    role="combobox"
    style={Object {}}
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-cascader-input-wrapper muya-input-input-wrapper"
      disabled={true}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      size="m"
      style={Object {}}
    >
      <input
        aria-describedby={null}
        className="c3 c4 muya-input-input"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="请选择"
        readOnly={true}
        size="m"
        type="text"
        value=""
      />
      <div
        className="c5 muya-input-node-wrapper"
        size="m"
      >
        <div
          className="c6"
        >
          <svg
            aria-hidden="false"
            className="c7"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <span
    aria-describedby={null}
    disabled={true}
  >
    [切换城市]
  </span>,
  .c10 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c9 {
  margin-left: 6px;
  cursor: pointer;
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 10px;
  height: 10px;
  color: rgba(204,204,204,1);
}

.c9:hover {
  color: rgba(191,191,191,1);
}

.c9:active {
  color: rgba(179,179,179,1);
}

.c9 svg {
  width: 10px;
  height: 10px;
}

.c7 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: none;
  color: rgba(102,102,102,1);
  background-color: rgba(248,248,248,1);
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  padding: 0 8px;
  margin-right: 8px;
  font-size: 12px;
  line-height: 1;
  height: 24px;
  white-space: nowrap;
  border-radius: 2px;
  cursor: default;
  vertical-align: middle;
  margin: 0 4px 4px 0;
  height: 22px;
  line-height: 22px;
}

.c7 .c8 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  overflow-x: hidden;
  overflow-y: auto;
  outline: none;
}

.c5 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  display: inline-block;
  overflow: hidden;
  background: transparent;
  border-width: 0;
  margin: 0;
  padding: 0;
  outline: 0;
  margin-bottom: 4px;
}

.c14 {
  position: absolute;
  top: 0;
  left: 0;
  visibility: hidden;
  height: 0;
  overflow: scroll;
  white-space: pre;
  line-height: 1;
}

.c16 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  margin: auto;
}

.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  overflow: hidden;
  outline: none;
  box-sizing: border-box;
  width: 240px;
  border: 1px solid rgba(221,221,221,1);
  box-shadow: none;
  border-radius: 2px;
  background: rgba(255,255,255,1);
  line-height: 1;
  font-size: 12px;
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  cursor: text;
  padding: 4px 4px 0 4px;
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c2 .c3 {
  padding-left: 12px;
  padding-right: 24px;
  max-height: 63px;
  padding-left: 0;
}

.c2 .c3::-webkit-scrollbar {
  width: 4px;
  height: 4px;
}

.c2 .c3::-webkit-scrollbar-thumb {
  border-radius: 999rem;
  background-color: transparent;
}

.c2 .c3:hover::-webkit-scrollbar-thumb {
  background-color: rgba(51,51,51,0.12);
}

.c2 .c3:hover::-webkit-scrollbar-thumb:hover {
  background-color: rgba(51,51,51,0.16);
}

.c2 .c3::-webkit-scrollbar-button:vertical:end:decrement {
  display: block;
  width: 4px;
  height: 4px;
  background-color: transparent;
}

.c2 .c15 {
  right: 12px;
}

.c2 .c11 {
  width: 1px;
  color: rgba(102,102,102,1);
}

.c2 input[readonly] {
  cursor: pointer;
}

.c2 input::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input::placeholder {
  color: rgba(204,204,204,1);
}

.c2 .c13 {
  font-size: 12px;
}

.c2 .c6 {
  max-width: 208px;
}

.c18 .c19,
.c18 .c1 {
  cursor: pointer;
}

.c18 .c19 input,
.c18 .c1 input,
.c18 .c19 svg,
.c18 .c1 svg {
  cursor: pointer;
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 .c20 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0 .c19,
.c0 .c1 {
  cursor: pointer;
}

.c0 .c19 input,
.c0 .c1 input,
.c0 .c19 svg,
.c0 .c1 svg {
  cursor: pointer;
}

.c17 {
  line-height: 0;
  -webkit-animation: bFXLB 150ms linear;
  animation: bFXLB 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

<div
    className="c0 muya-cascader-input-wrapper"
    id="test"
    role="combobox"
    style={Object {}}
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-cascader-input-wrapper muya-tags-input-input-wrapper"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3 c4 muya-tags-input-content-wrapper"
      >
        <div
          className="c5 muya-tags-input-tags-wrapper"
        >
          <span
            className="c6 c7 muya-tags-input-tag"
            onClick={[Function]}
            style={
              Object {
                "opacity": 1,
                "transitionDuration": "300ms",
                "transitionProperty": "opacity",
                "transitionTimingFunction": "cubic-bezier(0.0, 0, 0.2, 1)",
              }
            }
          >
            <span
              className="c8 muya-tag-children-wrapper"
            >
              福建/泉州
            </span>
            <span
              className="c9 muya-tag-close-icon"
            >
              <svg
                aria-hidden="false"
                className="c10"
                focusable="false"
                onClick={[Function]}
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
                />
              </svg>
            </span>
          </span>
          <input
            aria-describedby={null}
            autoFocus={false}
            className="c11 c12 muya-tags-input-input"
            disabled={false}
            id="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            placeholder=""
            readOnly={true}
            type="text"
            value=""
          />
          <div
            className="c13 c14"
          >
            
          </div>
        </div>
      </div>
      <div
        className="c15 c16"
      >
        <div
          className="c17"
        >
          <svg
            aria-hidden="false"
            className="c10"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  .c10 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c9 {
  margin-left: 6px;
  cursor: pointer;
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 10px;
  height: 10px;
  color: rgba(204,204,204,1);
}

.c9:hover {
  color: rgba(191,191,191,1);
}

.c9:active {
  color: rgba(179,179,179,1);
}

.c9 svg {
  width: 10px;
  height: 10px;
}

.c7 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: none;
  color: rgba(102,102,102,1);
  background-color: rgba(248,248,248,1);
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  transition: all 150ms cubic-bezier(0.4,0,0.2,1);
  padding: 0 8px;
  margin-right: 8px;
  font-size: 12px;
  line-height: 1;
  height: 24px;
  white-space: nowrap;
  border-radius: 2px;
  cursor: default;
  vertical-align: middle;
  margin: 0 4px 4px 0;
  height: 22px;
  line-height: 22px;
}

.c7 .c8 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c4 {
  position: relative;
  box-sizing: border-box;
  overflow-x: hidden;
  overflow-y: auto;
  outline: none;
}

.c5 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 {
  display: inline-block;
  overflow: hidden;
  background: transparent;
  border-width: 0;
  margin: 0;
  padding: 0;
  outline: 0;
  margin-bottom: 4px;
}

.c14 {
  position: absolute;
  top: 0;
  left: 0;
  visibility: hidden;
  height: 0;
  overflow: scroll;
  white-space: pre;
  line-height: 1;
}

.c16 {
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  margin: auto;
}

.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  overflow: hidden;
  outline: none;
  box-sizing: border-box;
  width: 240px;
  border: 1px solid rgba(221,221,221,1);
  box-shadow: none;
  border-radius: 2px;
  background: rgba(255,255,255,1);
  line-height: 1;
  font-size: 12px;
  -webkit-transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  transition: border 150ms cubic-bezier(0.4,0,0.2,1);
  cursor: text;
  padding: 4px 4px 0 4px;
}

.c2:hover {
  border-color: rgba(209,209,209,1);
}

.c2 .c3 {
  padding-left: 12px;
  padding-right: 24px;
  max-height: 63px;
  padding-left: 0;
}

.c2 .c3::-webkit-scrollbar {
  width: 4px;
  height: 4px;
}

.c2 .c3::-webkit-scrollbar-thumb {
  border-radius: 999rem;
  background-color: transparent;
}

.c2 .c3:hover::-webkit-scrollbar-thumb {
  background-color: rgba(51,51,51,0.12);
}

.c2 .c3:hover::-webkit-scrollbar-thumb:hover {
  background-color: rgba(51,51,51,0.16);
}

.c2 .c3::-webkit-scrollbar-button:vertical:end:decrement {
  display: block;
  width: 4px;
  height: 4px;
  background-color: transparent;
}

.c2 .c15 {
  right: 12px;
}

.c2 .c11 {
  width: 1px;
  color: rgba(102,102,102,1);
}

.c2 input[readonly] {
  cursor: pointer;
}

.c2 input::-webkit-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input::-moz-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input:-ms-input-placeholder {
  color: rgba(204,204,204,1);
}

.c2 input::placeholder {
  color: rgba(204,204,204,1);
}

.c2 .c13 {
  font-size: 12px;
}

.c2 .c6 {
  max-width: 208px;
}

.c18 .c19,
.c18 .c1 {
  cursor: pointer;
}

.c18 .c19 input,
.c18 .c1 input,
.c18 .c19 svg,
.c18 .c1 svg {
  cursor: pointer;
}

.c0 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 .c20 {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c0 .c19,
.c0 .c1 {
  cursor: pointer;
}

.c0 .c19 input,
.c0 .c1 input,
.c0 .c19 svg,
.c0 .c1 svg {
  cursor: pointer;
}

.c17 {
  line-height: 0;
  -webkit-animation: bFXLB 150ms linear;
  animation: bFXLB 150ms linear;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  cursor: pointer;
}

<div
    className="c0 muya-cascader-input-wrapper custom-input-wrapper"
    id="test"
    role="combobox"
    style={Object {}}
    tabIndex={0}
  >
    <div
      className="c1 c2 muya-cascader-input-wrapper custom-input-wrapper muya-tags-input-input-wrapper"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={Object {}}
    >
      <div
        className="c3 c4 muya-tags-input-content-wrapper"
      >
        <div
          className="c5 muya-tags-input-tags-wrapper"
        >
          <span
            className="c6 c7 muya-tags-input-tag"
            onClick={[Function]}
            style={
              Object {
                "opacity": 1,
                "transitionDuration": "300ms",
                "transitionProperty": "opacity",
                "transitionTimingFunction": "cubic-bezier(0.0, 0, 0.2, 1)",
              }
            }
          >
            <span
              className="c8 muya-tag-children-wrapper"
            >
              福建/泉州
            </span>
            <span
              className="c9 muya-tag-close-icon"
            >
              <svg
                aria-hidden="false"
                className="c10"
                focusable="false"
                onClick={[Function]}
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
                />
              </svg>
            </span>
          </span>
          <span
            className="c6 c7 muya-tags-input-tag"
            onClick={[Function]}
            style={
              Object {
                "opacity": 1,
                "transitionDuration": "300ms",
                "transitionProperty": "opacity",
                "transitionTimingFunction": "cubic-bezier(0.0, 0, 0.2, 1)",
              }
            }
          >
            <span
              className="c8 muya-tag-children-wrapper"
            >
              1+
            </span>
            <span
              className="c9 muya-tag-close-icon"
            >
              <svg
                aria-hidden="false"
                className="c10"
                focusable="false"
                onClick={[Function]}
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M512 421.653L871.04 62.592l90.517 90.507-359.04 359.04 358.87 358.901-90.507 90.517L512 602.667 154.507 960.16 64 869.643 421.493 512.16 63.84 154.507 154.357 64 512 421.653z"
                />
              </svg>
            </span>
          </span>
          <input
            aria-describedby={null}
            autoFocus={false}
            className="c11 c12 muya-tags-input-input"
            disabled={false}
            id="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            placeholder=""
            readOnly={true}
            type="text"
            value=""
          />
          <div
            className="c13 c14"
          >
            
          </div>
        </div>
      </div>
      <div
        className="c15 c16"
      >
        <div
          className="c17"
        >
          <svg
            aria-hidden="false"
            className="c10"
            focusable="false"
            style={
              Object {
                "color": "rgba(204,204,204,1)",
                "height": 8,
                "width": 8,
              }
            }
            viewBox="0 0 1024 1024"
          >
            <path
              d="M832.267 277.333H189.46a21.333 21.333 0 00-17.685 33.259l306.09 453.76a21.333 21.333 0 0034.817.779L849.387 311.36a21.333 21.333 0 00-17.12-34.027z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
